keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,669 @@
+
+  This feature is incompatible with supporting Pod replacements without knowing the behavior of a
+  parent controller for the Pods.
+
+- Support for partial-admission.
+
+  Since all pods are already created, an implementation of partial admission would imply the
+  deletion of some pods. It is not clear if this matches users expectations, as opposed to support
+  for elastic groups.
+
+- Support elastic groups of Pods, where the number of Pods changes after the job started.
+
+  While these jobs are one of the motivations for this KEP, the current proposal doesn't support
+  them. These jobs can be addressed in follow up KEPs.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+#### Story 1
+
+As an platform developer, I can queue plain Pods. I just add a queue name to the Pods through a
+label.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: foo
+  namespace: pod-namespace
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+#### Story 2
+
+As a platform developer, I can queue groups of Pods that share the same shape (Pod specs).
+In addition to the queue name, I can specify how many Pods belong to the group.
+
+<<[UNRESOLVED usability of labels and annotations ]>>
+Could the split among labels and annotations cause user mistakes?
+Note that integers cannot be used as label values.
+<<[/UNRESOLVED]>>
+
+The pods of a job with a single spec (similar to Pods in an Indexed Job), look like follows:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-index-0
+  namespace: pod-namespace
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+  annotations:
+    kueue.x-k8s.io/pod-group-count: "10"
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+#### Story 3
+
+As a platform developer, I can queue groups of Pods that have multiple shapes.
+In addition to the queue name, I can specify how many shapes to expect for the group and how
+many Pods are expected for the shape.
+
+The pods of a job following a driver-workers paradigm look like follows:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-driver
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+    kueue.x-k8s.io/pod-group-role: driver
+  annotations:
+    kueue.x-k8s.io/pod-group-roles: "2"
+    kueue.x-k8s.io/pod-group-role-count: "1"
```

Comment:
Should we limit the number of `pod-group-role-count` per the same `pod-group-name` to `8`, similar to the limit of the workload podSets?

https://github.com/kubernetes-sigs/kueue/blob/acec9e1bd99f0a51e555abfe42692d52adc216a1/apis/kueue/v1beta1/workload_types.go#L31-L35

IIUC, the `pod-group-role` concept is the same as Workload PodSet, right?

Reply:
Yes, added these details in the Design section.